### ISSUE TITLE
Fix false negatives for some rules when using CSS-in-JS parser

### ISSUE THREAD

**Issue by ark120202** [stylelint/stylelint#3930]
> Clearly describe the bug

Some rules don't work with CSS-in-JS parser

> Which rule, if any, is the bug related to?

- `property-no-unknown`
- `value-no-vendor-prefix`
- `max-nesting-depth`

(these are just few random ones I've tested, but probably there are more)

> What CSS is needed to reproduce the bug?

```css
a {
  rule: 1;
}
```

```js
const Component = styled.a`
  rule: 1;
`
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "property-no-unknown": true
  }
}
```

> Which version of stylelint are you using?

`9.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint test.{css,js}`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, `css-in-js`

> What did you expect to happen?

```shell
test.css
 2:3  ×  Unexpected unknown property "rule"   property-no-unknown

test.js
 2:3  ×  Unexpected unknown property "rule"   property-no-unknown
```

> What actually happened (e.g. what warnings or errors did you get)?

```shell
test.css
 2:3  ×  Unexpected unknown property "rule"   property-no-unknown
```

---

This is happening because `postcss-jsx` produces different AST from regular css, which appears to be expected (#3875), making [isStandardSyntaxDeclaration](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxDeclaration.js) return `false` and `property-no-unknown` skipping the declaration.

Aside from core rules, I've seen it in `stylelint-order` plugin, which uses it's own utility function that also not handles this case. I think there would be other plugins like this. If it's impossible to make this parser generate similar AST, I think it would be useful to document it that plugins also should be tested with it's output.

It also would be good to test all core rules for compatibility with that AST.

**Comment by gucong3000** [comment]
> This appears to be a regression introduced by #3875

It seems that there is no connection.

The same problem should happen on HTML:

```html
<a style="rule: 1;">
```
